src/performance.js

```diff
@@ -58,8 +66,48 @@ export class Performance {
 
     /** @const @private {!Array<TickEventDef>} */
     this.events_ = [];
+
+    /** @private {boolean} */
+    this.isViewerVisible_ = false;
+
+    this.win.addEventListener('unload', this.flush.bind(this));
```

Comment:
Yes, either plus periodically or on `visibilitychange`.

Maybe AMP should dispatch something like unload. Amp-analytics also needs it.

`unload` should be forbidden in presubmit! We never want to register it because it kills fast back navigation.